src/components/mobile-nav-item/mobile-nav-item.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from 'react'
+import type { FC } from 'react'
+
+import { useId } from '../../storybook/random-id'
+
+import { Icon } from '../icon'
+
+import { MobileNavItemExpandableProps, MobileNavItemSimpleProps, MobileNavSubItemProps } from './types'
+import {
+  ElMobileNavItemAnchor,
+  ElMobileNavItemBadge,
+  ElMobileNavItemContent,
+  ElMobileNavItemExpanderButton,
+  ElMobileNavItemListItem,
+  ElMobileNavItemUnorderedList,
+  ElMobileNavSubItemUnorderedList,
+} from './styles'
+import { MobileNavSubItem } from './mobile-nav-sub-item'
+
+export type MobileNavItemProps = MobileNavItemSimpleProps | MobileNavItemExpandableProps
+
+const isExpandableVariant = (props: MobileNavItemProps): props is MobileNavItemExpandableProps => {
+  return 'children' in props
+}
+
+type MobileNavItemFC = FC<MobileNavItemProps> & {
+  SubItem: FC<MobileNavSubItemProps>
+}
+
+const MobileNavItem: MobileNavItemFC = (props) => {
+  const [isExpanded, setIsExpanded] = useState(Boolean(props?.isDefaultExpanded))
+  const panelId = useId()
+
+  if (isExpandableVariant(props)) {
+    const { isActive, label, hasBadge, children } = props ?? {}
+
+    return (
+      <ElMobileNavItemUnorderedList data-is-expanded={isExpanded} role="menubar" aria-label={label}>
```

Comment:
**issue:** I'm not sure `menubar` is the correct role to use here. If we're trying to follow the [Disclosure pattern](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation/), it doesn't use this role at all...

Reply:
> I'm not sure menubar is the correct role to use here. If we're trying to follow the [Disclosure pattern](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation/), it doesn't use this role at all...

Ah yes, I got mixed up while trying to research between `Disclosure` and `Menu` pattern, apologies I'll give the update 🤦